pennylane/qinfo/fisher.py

```diff
@@ -0,0 +1,237 @@
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classical and quantum fisher information matrices"""
+# pylint: disable=import-outside-toplevel, not-callable
+import functools
+import pennylane as qml
+
+from pennylane.transforms import batch_transform
+
+# TODO: create qml.math.jacobian and replace it here
+def _torch_jac(circ):
+    """Torch jacobian as a callable function"""
+    import torch
+
+    def wrapper(*args, **kwargs):
+        loss = functools.partial(circ, **kwargs)
+        if len(args) > 1:
+            return torch.autograd.functional.jacobian(loss, (args))
+        return torch.autograd.functional.jacobian(loss, *args)
+
+    return wrapper
+
+
+# TODO: create qml.math.jacobian and replace it here
+def _tf_jac(circ):
+    """TF jacobian as a callable function"""
+    import tensorflow as tf
+
+    def wrapper(*args, **kwargs):
+        with tf.GradientTape() as tape:
+            loss = circ(*args, **kwargs)
+        return tape.jacobian(loss, (args))
+
+    return wrapper
+
+
+def classical_fisher(qnode, argnums=0):
+    r"""Returns a function that computes the classical fisher information matrix (CFIM) of a given :class:`.QNode` or quantum tape.
+
+    Given a parametrized (classical) probability distribution :math:`p(\bm{\theta})`, the classical fisher information matrix quantifies how changes to the parameters :math:`\bm{\theta}`
+    are reflected in the probability distribution. For a parametrized quantum state, we apply the concept of classical fisher information to the computational
+    basis measurement.
+    More explicitly, this function implements eq. (15) in `arxiv:2103.15191 <https://arxiv.org/abs/2103.15191>`_:
+
+    .. math::
+
+        \text{CFIM}_{i, j} = \sum_{\ell=0}^{2^N-1} \frac{1}{p_\ell(\bm{\theta})} \frac{\partial p_\ell(\bm{\theta})}{\partial \theta_i} \frac{\partial p_\ell(\bm{\theta})}{\partial \theta_j}
+
+    for :math:`N` qubits.
+
+    Args:
+        tape (:class:`.QNode` or qml.QuantumTape): A :class:`.QNode` or quantum tape that may have arbitrary return types.
+
+    Returns:
+        func: The function that computes the classical fisher information matrix. This function accepts the same signature as the :class:`.QNode`. If the signature contains one differentiable
+        variable ``params``, the function returns a matrix of size ``(len(params), len(params))``. For multiple differentiable arguments ``x, y, z``, it returns a list of sizes
+        ``[(len(x), len(x)), (len(y), len(y)), (len(z), len(z))]``.
+
+    .. warning::
+
+        In its current form, this functionality is not hardware compatible and can only be used by simulators.
+        It is currently also not differentiable.
```

Comment:
I thought this was hardware compatible? 🙂

Reply:
Differentiating through a QNode with `qml.probs()` should work on hardware

Reply:
Indeed! Added a test for it and removed the statement :)